### DISCUSSION TITLE
Como lidar com delete de um registro do banco da qual outra tabela depende?

### DISCUSSION THREAD

**Discussion by oimarcella** [balta-io/forum#563] [Tira Dúvidas] [answered]
Estou fazendo os desafios do curso de Dapper e implementei o método para adicionar um role a um usuário.
![image](https://github.com/balta-io/forum/assets/43733159/c40a1ef9-6993-48db-b189-e97508d41630)

Como eu poderia implementar o delete de role, se usuários dependem das roles?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @oimarcella, como vai?

Que tal você esboçar algo e postar aqui, e eu te guio se está ou não no caminho certo?

**Reply by oimarcella** [reply]
Sim, vou pensar nisso então.
Porém a princípio, penso em 2 opções.

1 - Não permitir que uma role seja deletada se ela ja estiver associada a outro registro. Nesse caso o registro dependente precisaria ser deletado primeiro.
2 - Remover a role deletada do registro dos usuários que a tem.

E para de fato colocar isso em código, vou pensar.
Não me parece nada viável percorrer listas de usuários, possivelmente lista longa, e ir removendo um por um.
